codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieHeaderNames.java

```diff
@@ -28,6 +28,39 @@
 
     public static final String HTTPONLY = "HTTPOnly";
 
+    public static final String SAMESITE = "SameSite";
+
+    /**
+     * Possible values for the SameSite attribute.
+     * See <a href=https://tools.ietf.org/html/draft-west-cookie-incrementalism-00#section-3.1>changes to RFC6265bis</a>
+     */
+    public enum SameSite {
+        Lax,
+        Strict,
+        None;
+
+        /**
+         * Return the enum value corresponding to the passed in same-site-flag, using a case insensitive comparison.
+         *
+         * @param name value for the SameSite Attribute
+         * @return enum value for the provided name or null
+         */
+        static SameSite of(String name) {
+            if (name != null) {
+                try {
+                    return SameSite.valueOf(name);
+                } catch (IllegalArgumentException e) {
+                    for (SameSite each : SameSite.class.getEnumConstants()) {
```

Comment:
Why do you want to do that?
From https://tools.ietf.org/html/draft-west-first-party-cookies-07#section-4.1:

> If "cookie-av"'s "attribute-value" is not a **case-sensitive** match
       for "Strict" or "Lax", ignore the "cookie-av".

I'd say to initially stick to the spec and not allow for broken implementations that don't use the proper case.

Reply:
@slandelle +1 ... @dvlato  lets just return `null` in the case

Reply:
Hi,

The new draft for 'incrementally better cookies' removes the case sensitive restriction for None values:
   2.  If cookie-av's attribute-value is a **case-insensitive**
       match for "None", set "enforcement" to "None".

The status of those drafts is not perfect but I would expect an updated version to say "case-insensitive" match wherever it says case sensitive match now.  

If you don't want to go down the "full case-insensitive"  route,  should I add case insensitivity comparisons just for the None value?

Reply:
@dvlato It looks like https://tools.ietf.org/html/draft-west-cookie-incrementalism-00 from May 7, 2019, expired on November 8, 2019 was superseded by https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-05 from February 5, 2020, expiring on August 8, 2020.

From section 5.4:

>    13.  If the cookie-attribute-list contains an attribute with an attribute-name of "SameSite", set the cookie's same-site-flag to the attribute-value of the last attribute in the cookie-attribute-list with an attribute-name of "SameSite" (i.e. either "Strict", "Lax", or "None").  Otherwise, set the cookie's same-site-flag to "None".

I read this as strict equality, hence case sensitive.

Reply:
Oh, I hadn't noticed there was a superseding draft. Thanks for that!
From the same draft you mention:
https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-05#section-5.3.7

If cookie-av's attribute-value is a case-insensitive match for
       "Strict", set "enforcement" to "Strict".

   3.  If cookie-av's attribute-value is a case-insensitive match for
       "Lax", set "enforcement" to "Lax".

Reply:
Nice, I missed that :)

That settles it then, case insensitive it is.